app/cp_nigeria/views.py

```diff
@@ -1505,14 +1505,21 @@ def save_graph_to_db(request, proj_id):
             graph_id = "stacked_timeseries"
         attr_name = f"{graph_id}_graph"
         image_url = request.POST.get("image_url")
-        report_qs = ImplementationPlanContent.objects.filter(simulation=project.scenario.simulation)
-        if report_qs.exists():
-            report_content = report_qs.first()
-            setattr(report_content, attr_name, image_url)
-            report_content.save()
-            return JsonResponse({"status": "saved " + attr_name + " to db"})
-        else:
-            return JsonResponse({"status": "db object could not be found"})
+
+        try:
+            with transaction.atomic():
+                report_qs = ImplementationPlanContent.objects.select_for_update().filter(
+                    simulation=project.scenario.simulation
+                )
+                if report_qs.exists():
+                    report_content = report_qs.first()
```

Comment:
We should only have one right? By using `.first()` you will silence the potential future bug. I think it is good that the GUI doesn't get a 500 Error but it would be good to use the logging to the server log file, you can simply use `logging.error`

Reply:
Do you mean I should be using `get()` here instead? Or leaving it like this but using `logging.error` when the query returns more than one object?

Reply:
using `logging.error` and leaving like this :)